data/generator/sfDoctrineRestGenerator/default/template/actions/actions.class.php

```diff
@@ -50,7 +50,13 @@ class <?php echo $this->getGeneratedModuleName() ?>Actions extends <?php echo $t
 
 <?php include dirname(__FILE__).'/../../parts/postValidate.php' ?>
 
+<?php $mode = '' ?>
 <?php include dirname(__FILE__).'/../../parts/query.php' ?>
+<?php $mode = sfDoctrineRestGenerator::GET_MODE_LIST ?>
+<?php include dirname(__FILE__).'/../../parts/query.php' ?>
+<?php $mode = sfDoctrineRestGenerator::GET_MODE_SHOW ?>
+<?php include dirname(__FILE__).'/../../parts/query.php' ?>
```

Comment:
Maybe it is not the best solution working like this, but it reduces code duplication a lot...